### ISSUE TITLE
Sysvar documentation is incorrect

### ISSUE THREAD

**Issue by morgo** [pingcap/docs#11086]
## Error Report

I ran a script to compare the current sysvar docs to the source code. I found the following mistakes:

```
Still missing docs:

tidb_enable_ddl (I verified: it is in the 6.3 source)

Scope is wrong:

error_count
tidb_config
tidb_enable_prepared_plan_cache
tidb_log_file_max_days
tidb_prepared_plan_cache_size
tidb_row_format_version

Default value is wrong:

tidb_allow_function_for_expression_index
tidb_enable_paging
tidb_enable_tmp_storage_on_oom
tidb_guarantee_linearizability
tidb_mem_quota_analyze
tidb_memory_usage_alarm_ratio
tidb_stats_load_pseudo_timeout (also says it's experimental and you shouldn't enable it - but its on by default.)

Range is wrong:

tidb_max_paging_size
tidb_min_paging_size
tidb_opt_concurrency_factor
tidb_opt_copcpu_factor
tidb_opt_correlation_exp_factor
tidb_opt_desc_factor
tidb_opt_disk_factor

Not in alphabetical order:

tidb_generate_binary_plan
tidb_max_auto_analyze_time
tidb_max_tiflash_threads
tidb_mpp_store_fail_ttl
tidb_opt_tiflash_concurrency_factor
tidb_opt_force_inline_cte 


Mistakes:

- tidb_batch_commit, tidb_batch_delete, tidb_batch_insert, tidb_last_ddl_info, tidb_last_plan_replayer_token, tidb_last_query_info, tidb_last_txn_info, tidb_memory_debug_mode_alarm_ratio, tidb_read_consistency, tx_read_ts, txn_scope list "PERSISTS to cluster: NO". This is only relevant to GLOBAL scoped variables which makes it confusing to include.
- tidb_batch_pending_tiflash_count lists the scope as `[0, 2147483647]` (thats the range.)
- docs for tidb_optimizer_selectivity_level are just wrong: scope, default value, range.
- tidb_stats_load_pseudo_timeout says it's experimental and you shouldn't enable it - but its on by default. The docs are wrong.
```

Please answer the following questions before submitting your issue. Thanks!

1. What is the URL/path of the document related to this issue?

https://docs.pingcap.com/tidb/stable/system-variables

2. How would you like to improve it?

**Comment by TomShawn** [comment]
Hi, thanks for checking the incorrect information in the TiDB system variable doc. I went through all variables you mentioned by using the `information_schema.variables_info` table (for v6.2.0 and v6.3.0 respectively). Almost all the variables you mentioned have incorrect doc information, except for the following ones:

- The default value of `tidb_enable_paging` is `OFF`, just as the documentation says.
- The default value of `tidb_memory_usage_alarm_ratio` is `0.8`, just as the documentation says.
- The default value of `tidb_enable_tmp_storage_on_oom` is `ON`, just as [this PR](https://github.com/pingcap/docs/pull/10910/files#diff-ec2cbb115cc58fe3afe2862eb5148732d323cbf7ae35dea97bd381a1f8d0e8a0R1589) says, which will be published along with v6.4.

For the missing `tidb_enable_ddl`, it is documented in [this PR](https://github.com/pingcap/docs/pull/10910/files#diff-ec2cbb115cc58fe3afe2862eb5148732d323cbf7ae35dea97bd381a1f8d0e8a0R1197) and will be published along with v6.4.

For all the other variables, the incorrect information is fixed in https://github.com/pingcap/docs/pull/11100 and I'll make version-specific to changes once it is cherry-picked to v6.2 or v6.1. Would you please take a look? Thanks~

**Comment by morgo** [comment]
It LGTM so far. I'll take another look after #10910 and #11100 merge.

**Comment by morgo** [comment]
It's much closer now, which makes it easier to understand diffs.

If I diff the docs in master to the release-6.3 branch the following 15 items are not documented:
```
tidb_enable_ddl
tidb_enable_tmp_storage_on_oom
tidb_opt_concurrency_factor_v2
tidb_opt_copcpu_factor_v2
tidb_opt_cpu_factor_v2
tidb_opt_desc_factor_v2
tidb_opt_disk_factor_v2
tidb_opt_memory_factor_v2
tidb_opt_network_factor_v2
tidb_opt_range_max_size
tidb_opt_scan_factor_v2
tidb_opt_seek_factor_v2
tidb_opt_tiflash_concurrency_factor
tidb_opt_tiflash_cpu_factor_v2
tidb_opt_tiflash_scan_factor_v2
```

**Comment by TomShawn** [comment]
> It's much closer now, which makes it easier to understand diffs.
> 
> If I diff the docs in master to the release-6.3 branch the following 15 items are not documented:
> 
> ```
> tidb_enable_ddl
> tidb_enable_tmp_storage_on_oom
> tidb_opt_concurrency_factor_v2
> tidb_opt_copcpu_factor_v2
> tidb_opt_cpu_factor_v2
> tidb_opt_desc_factor_v2
> tidb_opt_disk_factor_v2
> tidb_opt_memory_factor_v2
> tidb_opt_network_factor_v2
> tidb_opt_range_max_size
> tidb_opt_scan_factor_v2
> tidb_opt_seek_factor_v2
> tidb_opt_tiflash_concurrency_factor
> tidb_opt_tiflash_cpu_factor_v2
> tidb_opt_tiflash_scan_factor_v2
> ```

- Hi, these variables are in the v6.3 doc and in the v6.3 code. You can find them right on [this page](https://docs.pingcap.com/tidb/v6.3/system-variables). However, these variables are removed in [the code of the master branch](https://github.com/pingcap/tidb/pull/38324). So I do not add them to the TiDB dev doc.

    ```
    tidb_opt_concurrency_factor_v2
    tidb_opt_copcpu_factor_v2
    tidb_opt_cpu_factor_v2
    tidb_opt_desc_factor_v2
    tidb_opt_disk_factor_v2
    tidb_opt_memory_factor_v2
    tidb_opt_network_factor_v2
    tidb_opt_scan_factor_v2
    tidb_opt_seek_factor_v2
    tidb_opt_tiflash_cpu_factor_v2
    tidb_opt_tiflash_scan_factor_v2
    ```

- `tidb_opt_tiflash_concurrency_factor` was confirmed to be removed in code in master (by @qw4990) but I just found it in the latest v6.4.0-alpha-nightly build. I'll add it to https://github.com/pingcap/docs/pull/10910.
- `tidb_enable_ddl` and `tidb_enable_tmp_storage_on_oom` will be added to the dev and v6.3 doc once https://github.com/pingcap/docs/pull/10910 is merged. (I think it will be very soon.)
- `tidb_opt_range_max_size` is associated with a feature to be released in v6.4 (see https://github.com/pingcap/docs/pull/10935) although it was added to the v6.3 code. I'll cherry-pick https://github.com/pingcap/docs/pull/10935 to release-6.3 with version-specific changes once it is merged.

**Comment by morgo** [comment]
Here is a diff I ran against master showing the type/Default value annotation differences. In some cases it lacks consistency, in other cases it is wrong.

It's not completely definitive, since there are still some other cases where the range is wrong, but I needed to skip them for now:

```
diff --git a/system-variables.md b/system-variables.md
index d4c176fce..6f0588c62 100644
--- a/system-variables.md
+++ b/system-variables.md
@@ -111,7 +111,7 @@ mysql> SELECT * FROM t1;
 - Persists to cluster: Yes
 - Type: Enumeration
 - Default value: `aes-128-ecb`
-- Value options: `aes-128-ecb`, `aes-192-ecb`, `aes-256-ecb`, `aes-128-cbc`, `aes-192-cbc`, `aes-256-cbc`, `aes-128-ofb`, `aes-192-ofb`, `aes-256-ofb`, `aes-128-cfb`, `aes-192-cfb`, `aes-256-cfb`
+- Possible values: `aes-128-ecb`, `aes-192-ecb`, `aes-256-ecb`, `aes-128-cbc`, `aes-192-cbc`, `aes-256-cbc`, `aes-128-ofb`, `aes-192-ofb`, `aes-256-ofb`, `aes-128-cfb`, `aes-192-cfb`, `aes-256-cfb`
 - This variable sets the encryption mode for the built-in functions `AES_ENCRYPT()` and `AES_DECRYPT()`.
 
 ### character_set_client
@@ -198,7 +198,9 @@ mysql> SELECT * FROM t1;
 
 - Scope: GLOBAL
 - Persists to cluster: No, only applicable to the current TiDB instance that you are connecting to.
+- Type: Integer
 - Default value: `300`
+- Range: `[0, 2147483647]`
 - Unit: Milliseconds
 - Log DDL operations whose execution time exceeds the threshold value.
 
@@ -208,7 +210,7 @@ mysql> SELECT * FROM t1;
 - Persists to cluster: Yes
 - Type: Enumeration
 - Default value: `mysql_native_password`
-- Possible values: `mysql_native_password`, `caching_sha2_password`, and `tidb_sm3_password`
+- Possible values: `mysql_native_password`, `caching_sha2_password`, `tidb_sm3_password`
 - This variable sets the authentication method that the server advertises when the server-client connection is being established.
 - To authenticate using the `tidb_sm3_password` method, you can connect to TiDB using [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3).
 
@@ -254,12 +256,14 @@ For more possible values of this variable, see [Authentication plugin status](/s
 ### have_openssl
 
 - Scope: NONE
+- Type: Boolean
 - Default value: `DISABLED`
 - A read-only variable for MySQL compatibility. Set to `YES` by the server when the server has TLS enabled.
 
 ### have_ssl
 
 - Scope: NONE
+- Type: Boolean
 - Default value: `DISABLED`
 - A read-only variable for MySQL compatibility. Set to `YES` by the server when the server has TLS enabled.
 
@@ -303,7 +307,9 @@ This variable is an alias for [`last_insert_id`](#last_insert_id).
 ### last_insert_id
 
 - Scope: SESSION
+- Type: Integer
 - Default value: `0`
+- Range: `[0, 9223372036854775807]`
 - This variable returns the last `AUTO_INCREMENT` or `AUTO_RANDOM` value generated by an insert statement.
 - The value of `last_insert_id` is the same as the value returned by the function `LAST_INSERT_ID()`.
 
@@ -345,8 +351,9 @@ This variable is an alias for [`last_insert_id`](#last_insert_id).
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Default value: `67108864`
-- Range: `[1024, 1073741824]`. The value should be an integer multiple of 1024. If the value is not divisible by 1024, a warning will be prompted and the value will be rounded down. For example, when the value is set to 1025, the actual value in TiDB is 1024.
-- The maximum packet size allowed by the server and the client in one transmission of packets, in bytes.
+- Range: `[1024, 1073741824]`
+- The value should be an integer multiple of 1024. If the value is not divisible by 1024, a warning will be prompted and the value will be rounded down. For example, when the value is set to 1025, the actual value in TiDB is 1024.
+- The maximum packet size allowed by the server and the client in one transmission of packets.
 - This variable is compatible with MySQL.
 
 ### max_connections
@@ -596,6 +603,7 @@ mysql> SHOW GLOBAL VARIABLES LIKE 'max_prepared_stmt_count';
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Default value: `4096`
+- Range: `[0, 9223372036854775807]`
 - This variable is used to control the threshold at which the TiDB server prefers to send read requests to the replica in the same region as the TiDB server when [`tidb_replica_read`](#tidb_replica_read-new-in-v40) is set to `closest-adaptive`. If the estimated result is higher than or equal to this threshold, TiDB prefers to send read requests to the replica in the same region. Otherwise, TiDB sends read requests to the leader replica.
 
 ### tidb_allow_batch_cop <span class="version-mark">New in v4.0</span>
@@ -788,8 +796,9 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `4000`
-- Range: `[0, 2147483647]`
+- Range: `[0, 4294967295]`
 - Specifies the maximum number of permitted unavailable tables when you use `ALTER DATABASE SET TIFLASH REPLICA` to add TiFlash replicas. If the number of unavailable tables exceeds this limit, the operation will be stopped or setting TiFlash replicas for the remaining tables will be very slow.
 
 ### tidb_broadcast_join_threshold_count <span class="version-mark">New in v5.0</span>
@@ -816,7 +825,9 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `4`
+- Range: `[1, 256]`
 - Unit: Threads
 - This variable is used to set the concurrency of executing the `ANALYZE` statement.
 - When the variable is set to a larger value, the execution performance of other queries is affected.
@@ -850,7 +861,9 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 ### tidb_checksum_table_concurrency
 
 - Scope: SESSION
+- Type: Integer
 - Default value: `4`
+- Range: `[1, 256]`
 - Unit: Threads
 - This variable is used to set the scan index concurrency of executing the [`ADMIN CHECKSUM TABLE`](/sql-statements/sql-statement-admin-checksum-table.md) statement.
 - When the variable is set to a larger value, the execution performance of other queries is affected.
@@ -915,6 +928,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `ON`
 - This variable only applies to pessimistic transactions. For optimistic transactions, use [`tidb_constraint_check_in_place`](#tidb_constraint_check_in_place) instead.
 - When this variable is set to `OFF`, TiDB defers the unique constraint check of a unique index (to the next time when executing a statement that requires a lock to the index or to the time when committing the transaction). This helps improve performance but might be an unexpected behavior for some applications. See [Constraints](/constraints.md#pessimistic-transactions) for details.
@@ -973,8 +987,9 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `1`
-- Value options: `1`, `2`
+- Range: `[1, 2]`
 - TiDB v6.2.0 introduces the [Cost Model Version 2](/cost-model.md#cost-model-version-2), which is more accurate than the previous version in internal tests.
 - To enable the Cost Model Version 2, you can set the `tidb_cost_model_version` to `2`. If you set this variable to `1`, the Cost Model Version 1 will be used.
 - The version of cost model affects the plan decision of optimizer. For more details, see [Cost Model](/cost-model.md).
@@ -982,7 +997,9 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 ### tidb_current_ts
 
 - Scope: SESSION
+- Type: Integer
 - Default value: `0`
+- Range: `[0, 9223372036854775807]`
 - This variable is read-only. It is used to obtain the timestamp of the current transaction.
 
 ### tidb_ddl_disk_quota <span class="version-mark">New in v6.3.0</span>
@@ -1030,6 +1047,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `64`
 - Range: `[1, 256]`
 
@@ -1054,7 +1072,9 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 ### tidb_ddl_reorg_priority
 
 - Scope: SESSION
+- Type: Enumeration
 - Default value: `PRIORITY_LOW`
+- Possible values: `PRIORITY_LOW`, `PRIORITY_NORMAL`, `PRIORITY_HIGH`
 - This variable is used to set the priority of executing the `ADD INDEX` operation in the `re-organize` phase.
 - You can set the value of this variable to `PRIORITY_LOW`, `PRIORITY_NORMAL` or `PRIORITY_HIGH`.
 
@@ -1072,6 +1092,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Float
 - Default value: `0.8`
 - Range: `[0, 1]`
 - This variable is used to set the default selectivity of `like`, `rlike`, and `regexp` functions in the filter condition when estimating the number of rows. This variable also controls whether to enable TopN to help estimate these functions.
@@ -1245,7 +1266,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Type: Enumeration
-- Default value: `INT_ONLY`
+- Default value: `ON`
 - Possible values: `OFF`, `ON`, `INT_ONLY`
 - This variable is used to control whether to create the primary key as a [clustered index](/clustered-indexes.md) by default. "By default" here means that the statement does not explicitly specify the keyword `CLUSTERED`/`NONCLUSTERED`. Supported values are `OFF`, `ON`, and `INT_ONLY`:
     - `OFF` indicates that primary keys are created as non-clustered indexes by default.
@@ -1296,6 +1317,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `ON`
 - This variable controls whether to allow TiDB to use concurrent DDL statements. When concurrent DDL statements are used, the DDL execution flow is changed, and DDL statements are not easily blocked by other DDL statements. In addition, multiple indexes can be added at the same time.
 
@@ -1390,6 +1412,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `OFF`
 - This variable controls whether to enable the `FOREIGN KEY` feature.
 
@@ -1413,6 +1436,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `OFF`
 - This variable controls whether to enable the General Plan Cache feature.
 
@@ -1506,7 +1530,6 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Persists to cluster: Yes
 - Type: Boolean
 - Default value: `ON`
-- Value options: `OFF` and `ON`
 - TiDB v6.2.0 refactors the implementation of previous cost model. This variable controls whether to enable the refactored Cost Model implementation.
 - This variable is enabled by default because the refactored Cost Model uses the same cost formula as before, which does not change the plan decision.
 - If your cluster is upgraded from v6.1 to v6.2, this variable remains `OFF`, and it is recommended to enable it manually. If your cluster is upgraded from a version earlier than v6.1, this variable sets to `ON` by default.
@@ -1515,8 +1538,8 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `OFF`
-- Value options: `OFF` and `ON`
 - This variable controls the behavior when TiDB performs the `ONLY_FULL_GOUP_BY` check. For detailed information about `ONLY_FULL_GROUP_BY`, see the [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sqlmode_only_full_group_by). In v6.1.0, TiDB handles this check more strictly and correctly.
 - To avoid potential compatibility issues caused by version upgrades, the default value of this variable is `OFF` in v6.1.0.
 
@@ -1552,8 +1575,9 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 ### `tidb_enable_null_aware_anti_join` <span class="version-mark">New in v6.3.0</span>
 
-- Scope: SESSION ｜GLOBAL
+- Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `OFF`
 - This variable controls whether TiDB applies Null Aware Hash Join when ANTI JOIN is generated by subqueries led by special set operators `NOT IN` and `!= ALL`.
 
@@ -1561,6 +1585,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: In v6.1.0, the default value is `ON`. After v6.1.0, the default value is `OFF`.
 - Since v6.1.0, the [Join Reorder](/join-reorder.md) algorithm of TiDB supports Outer Join. This variable controls the support behavior. The default value is `OFF`, which means the Join Reorder's support for Outer Join is disabled by default.
 - For a cluster upgraded from a version earlier than v6.1.0, the default value is `OFF`. For a cluster upgraded from v6.1.0, the default value is `ON`.
@@ -1569,6 +1594,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `OFF`
 - Specifies whether to sort the final output result automatically.
 - For example, with this variable enabled, TiDB processes `SELECT a, MAX(b) FROM t GROUP BY a` as `SELECT a, MAX(b) FROM t GROUP BY a ORDER BY a, MAX(b)`.
@@ -1578,7 +1604,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Type: Boolean
-- Default value: `OFF`
+- Default value: `ON`
 - This variable controls whether to use the method of paging to send coprocessor requests in `IndexLookUp` operator.
 - User scenarios: For read queries that use `IndexLookup` and `Limit` and that `Limit` cannot be pushed down to `IndexScan`, there might be high latency for the read queries and high CPU usage for TiKV's `unified read pool`. In such cases, because the `Limit` operator only requires a small set of data, if you set `tidb_enable_paging` to `ON`, TiDB processes less data, which reduces query latency and resource consumption.
 - When `tidb_enable_paging` is enabled, for the `IndexLookUp` requests with `Limit` that cannot be pushed down and are fewer than `960`, TiDB uses the method of paging to send coprocessor requests. The fewer `Limit`, the more obvious the optimization.
@@ -1766,6 +1792,7 @@ Query OK, 0 rows affected (0.09 sec)
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `OFF`
 - This variable controls whether read requests in SQL write statements can be pushed down to TiFlash.
 
@@ -1958,9 +1985,11 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: GLOBAL
 - Persists to cluster: No, only applicable to the current TiDB instance that you are connecting to.
+- Type: Enumeration
 - Default value: `NO_PRIORITY`
+- Possible values: `NO_PRIORITY`, `LOW_PRIORITY`, `HIGH_PRIORITY`, `DELAYED`
 - This variable is used to change the default priority for statements executed on a TiDB server. A use case is to ensure that a particular user that is performing OLAP queries receives lower priority than users performing OLTP queries.
-- The default value `NO_PRIORITY` means that the priority for statements is not forced to change. Other options are `LOW_PRIORITY`, `DELAYED`, and `HIGH_PRIORITY` in ascending order.
+- The default value `NO_PRIORITY` means that the priority for statements is not forced to change.
 
 ### tidb_gc_concurrency <span class="version-mark">New in v5.0</span>
 
@@ -2000,10 +2029,11 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `86400`
 - Range: `[600, 31536000]`
 - Unit: Seconds
-- This variable is used to set the maximum time that active transactions block the GC safe point. During each time of GC, the safe point does not exceed the start time of the ongoing transactions by default. If the runtime of active transactions does not exceed this variable value, the GC safe point will be blocked until the runtime exceeds this value. This variable value is an integer type.
+- This variable is used to set the maximum time that active transactions block the GC safe point. During each time of GC, the safe point does not exceed the start time of the ongoing transactions by default. If the runtime of active transactions does not exceed this variable value, the GC safe point will be blocked until the runtime exceeds this value.
 
 ### tidb_gc_run_interval <span class="version-mark">New in v5.0</span>
 
@@ -2086,6 +2116,7 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `100`
 - Range: `[1, 100000]`
 - This variable controls the maximum number of execution plans that can be cached by General Plan Cache.
@@ -2094,6 +2125,7 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `ON`
 - This variable controls whether to generate binary-encoded execution plans in slow logs and statement summaries.
 - When this variable is set to `ON`, you can view visual execution plans in TiDB Dashboard. Note that TiDB Dashboard only provides visual display for execution plans generated after this variable is enabled.
@@ -2322,9 +2354,10 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `43200`
 - Range: `[0, 2147483647]`
-- Unit: seconds
+- Unit: Seconds
 - This variable is used to specify the maximum execution time of automatic `ANALYZE` tasks. When the execution time of an automatic `ANALYZE` task exceeds the specified time, the task will be terminated. When the value of this variable is `0`, there is no limit to the maximum execution time of automatic `ANALYZE` tasks.
 
 ### tidb_max_chunk_size
@@ -2360,8 +2393,10 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `-1`
 - Range: `[-1, 256]`
+- Unit: Threads
 - This variable is used to set the maximum concurrency for TiFlash to execute a request. The default value is `-1`, indicating that this system variable is invalid. When the value is `0`, the maximum number of threads is automatically configured by TiFlash.
 
 ### tidb_mem_oom_action <span class="version-mark">New in v6.1.0</span>
@@ -2395,7 +2430,9 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `-1`
+- Range: `[-1, 9223372036854775807]`
 - Unit: Bytes
 - This variable controls the maximum memory usage of TiDB updating statistics. Such a memory usage occurs when you manually execute [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) and when TiDB automatically analyzes tasks in the background. When the total memory usage exceeds this threshold, user-executed `ANALYZE` will exit, and an error message is reported that reminds you to try a lower sampling rate or retry later. If the automatic task in the TiDB background exits because the memory threshold is exceeded, and the sampling rate used is higher than the default value, TiDB will retry the update using the default sampling rate. When this variable value is negative or zero, TiDB does not limit the memory usage of both the manual and automatic update tasks.
 
@@ -2459,7 +2496,7 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 ### tidb_memory_debug_mode_min_heap_inuse
 
 - Scope: SESSION
-- Type: INT64
+- Type: Integer
 - Default value: `0`
 - This variable is used for the internal testing of TiDB. It is **NOT recommended** to set this variable. Enabling this variable will affect the performance of TiDB.
 - After configuring this parameter, TiDB will enter the memory debug mode to analyze the accuracy of memory tracking. TiDB will frequently trigger GC during the execution of subsequent SQL statements, and compare the actual memory usage and memory statistics. If the current memory usage is greater than `tidb_memory_debug_mode_min_heap_inuse` and the memory statistics error exceeds `tidb_memory_debug_mode_alarm_ratio`, TiDB will output the relevant memory information to the log and file.
@@ -2497,6 +2534,8 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
+- Range: `[1, 256]`
 - Default value: `1`
 - This variable sets the concurrency of the `MergeJoin` operator when a query is executed.
 - It is **NOT recommended** to set this variable. Modifying the value of this variable might cause data correctness issues.
@@ -2584,6 +2623,7 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `OFF`
 - This variable specifies whether to return an error immediately when the error occurs in a non-transactional DML statement.
 - When the value is set to `OFF`, the non-transactional DML statement stops immediately at the first error and returns the error. All the following batches are canceled.
@@ -2591,7 +2631,8 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 
 ### tidb_opt_agg_push_down
 
-- Scope: SESSION
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
 - Type: Boolean
 - Default value: `OFF`
 - This variable is used to set whether the optimizer executes the optimization operation of pushing down the aggregate function to the position before Join, Projection, and UnionAll.
@@ -2756,6 +2797,7 @@ mysql> desc select count(distinct a) from test.t;
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `0`
 - Range: `[0, 2147483647]`
 - This variable is used to control the selection of the TiDB Join Reorder algorithm. When the number of nodes participating in Join Reorder is greater than this threshold, TiDB selects the greedy algorithm, and when it is less than this threshold, TiDB selects the dynamic programming algorithm.
@@ -2896,13 +2938,14 @@ explain select * from t where age=5;
 </CustomContent>
 
 - Scope: SESSION
+- Type: Boolean
 - Default value: `OFF`
 - This variable is used to control whether to allow `INSERT`, `REPLACE`, and `UPDATE` statements to operate on the `_tidb_rowid` column. This variable can be used only when you import data using TiDB tools.
 
 ### tidb_optimizer_selectivity_level
 
 - Scope: SESSION
-- Persists to cluster: Yes
+- Type: Integer
 - Default value: `0`
 - Range: `[0, 2147483647]`
 - This variable controls the iteration of the optimizer's estimation logic. After changing the value of this variable, the estimation logic of the optimizer will change greatly. Currently, `0` is the only valid value. It is not recommended to set it to other values.
@@ -2911,7 +2954,9 @@ explain select * from t where age=5;
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Enumeration
 - Default value: `dynamic`
+- Possible values: `static`, `dynamic`, `static-only`, `dynamic-only`
 - Specifies whether to use `dynamic` or `static` mode for partitioned tables. Note that dynamic partitioning is effective only after full table-level statistics, or GlobalStats, are collected. Before GlobalStats are collected, TiDB will use the `static` mode instead. For detailed information about GlobalStats, see [Collect statistics of partitioned tables in dynamic pruning mode](/statistics.md#collect-statistics-of-partitioned-tables-in-dynamic-pruning-mode). For details about the dynamic pruning mode, see [Dynamic Pruning Mode for Partitioned Tables](/partitioned-table.md#dynamic-pruning-mode).
 
 ### tidb_persist_analyze_options <span class="version-mark">New in v5.4.0</span>
@@ -3022,7 +3067,7 @@ explain select * from t where age=5;
 > - This feature is incompatible with [`replica-read`](#tidb_replica_read-new-in-v40). Do not enable `tidb_rc_read_check_ts` and `replica-read` at the same time.
 > - If your client uses a cursor, it is not recommended to enable `tidb_rc_read_check_ts` in case that the previous batch of returned data has already been used by the client and the statement eventually fails.
 
-- Scope: INSTANCE. Since v6.3.0, the scope changes from GLOBAL or SESSION to INSTANCE.
+- Scope: GLOBAL
 - Persists to cluster: No, only applicable to the current TiDB instance that you are connecting to.
 - Type: Boolean
 - Default value: `OFF`
@@ -3037,6 +3082,7 @@ explain select * from t where age=5;
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Boolean
 - Default value: `OFF`
 - This variable is used to optimize the acquisition of timestamps and is suitable for scenarios with few point-write conflicts in `READ-COMMITTED` isolation level of pessimistic transactions. Enabling this variable can avoid the latency and overhead brought by obtaining the global timestamps during the execution of point-write statements. Currently, this variable is applicable to three types of point-write statements: `UPDATE`, `DELETE`, and `SELECT ...... FOR UPDATE`. A point-write statement refers to a write statement that uses the primary key or unique key as a filter condition and the final execution operator contains `POINT-GET`.
 - If the point-write conflicts are severe, enabling this variable will increase extra overhead and latency, resulting in performance regression. For details, see [Read Committed isolation level](/transaction-isolation-levels.md#read-committed-isolation-level).
@@ -3170,7 +3216,7 @@ explain select * from t where age=5;
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Type: Integer
-- Default value: `1`
+- Default value: `2`
 - Range: `[1, 2]`
 - Controls the format version of the newly saved data in the table. In TiDB v4.0, the [new storage row format](https://github.com/pingcap/tidb/blob/master/docs/design/2018-07-19-row-format.md) version `2` is used by default to save new data.
 - If you upgrade from a TiDB version earlier than 4.0.0 to 4.0.0, the format version is not changed, and TiDB continues to use the old format of version `1` to write data to the table, which means that **only newly created clusters use the new data format by default**.
@@ -3189,6 +3235,7 @@ explain select * from t where age=5;
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `9223372036854775807`
 - Range: `[1, 9223372036854775807]`
 - This variable controls the maximum number of continuous IDs to be allocated for the [`AUTO_RANDOM`](/auto-random.md) or [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) attribute. Generally, `AUTO_RANDOM` IDs or the `SHARD_ROW_ID_BITS` annotated row IDs are incremental and continuous in one transaction. You can use this variable to solve the hotspot issue in large transaction scenarios.
@@ -3315,7 +3362,7 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Type: Integer
-- Default value: `0`
+- Default value: `100`
 - Range: `[0, 2147483647]`
 - Unit: Milliseconds
 - This variable controls whether to enable the synchronously loading statistics feature. The default value `0` means that the feature is disabled. To enable the feature, you can set this variable to a timeout (in milliseconds) that SQL optimization can wait for at most to synchronously load complete column statistics. For details, see [Load statistics](/statistics.md#load-statistics).
@@ -3378,6 +3425,7 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 
 - Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `1`
 - This variable sets the concurrency of the `StreamAgg` operator when queries are executed.
 - It is **NOT recommended** to set this variable. Modifying the variable value might cause data correctness issues.
@@ -3490,7 +3538,7 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 
 ### tidb_track_aggregate_memory_usage
 
-- Scope: SESSION ｜ GLOBAL
+- Scope: SESSION | GLOBAL
 - Persists to cluster: Yes
 - Type: Boolean
 - Default value: `ON`
@@ -3540,6 +3588,7 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 
 - Scope: GLOBAL
 - Persists to cluster: Yes
+- Type: Integer
 - Default value: `16384`
 - Range: `[1, 1073741824]`
 - Unit: Bytes
@@ -3613,7 +3662,7 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 
 - Scope: SESSION | GLOBAL
 - Default value: `OFF`
-- Range: `ON | OFF`
+- Type: Boolean
 
 <CustomContent platform="tidb">
 
@@ -3625,13 +3674,15 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 
 - Scope: SESSION | GLOBAL
 - Default value: `8192`
-- Range: `[1, 18446744073709551616]`
+- Range: `[1, 18446744073709551615]`
 - When Fine Grained Shuffle is enabled, the window function pushed down to TiFlash can be executed in parallel. This variable controls the batch size of the data sent by the sender. The sender will send data once the cumulative number of rows exceeds this value.
 - Impact on performance: set a reasonable size according to your business requirements. Improper setting affects the performance. If the value is set too small, for example `1`,  it causes one network transfer per Block. If the value is set too large, for example, the total number of rows of the table, it causes the receiving end to spend most of the time waiting for data, and the piplelined computation can not work. To set a proper value, you can observe the distribution of the number of rows received by the TiFlash receiver. If most threads receive only a few rows, for example a few hundred, you can increase this value to reduce the network overhead.
 
 ### `tiflash_fine_grained_shuffle_stream_count` <span class="version-mark">New in v6.2.0</span>
 
 - Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Type: Integer
 - Default value: `0`
 - Range: `[-1, 1024]`
 - When the window function is pushed down to TiFlash for execution, you can use this variable to control the concurrency level of the window function execution. The possible values are as follows:
@@ -3681,7 +3732,7 @@ Internally, the TiDB parser transforms the `SET TRANSACTION ISOLATION LEVEL [REA
 ### tx_read_ts
 
 - Scope: SESSION
-- Default value: `0`
+- Default value: ""
 - In the Stale Read scenarios, this session variable is used to help record the Stable Read timestamp value.
 - This variable is used for the internal operation of TiDB. It is **NOT recommended** to set this variable.
```

**Comment by TomShawn** [comment]
> 

@morgo Hi, I do not quite understand your needs in https://github.com/pingcap/docs/issues/11086#issuecomment-1304613524. 

- Which version did you compare with the master branch?
- "type/Default value annotation differences. In some cases it lacks consistency, in other cases it is wrong."
    - Does this mean some default values and the types in the doc are wrong? I checked the master branch and found some default values are wrong. See https://github.com/pingcap/docs/pull/11261.
    - For "lacks consistency", does it mean the variable information is not consistent across versions?

**Comment by morgo** [comment]
I compared against master here. Some default values are wrong, but also range/type info is incorrect (less a problem, but still inconsistent). If you save the patch I pasted to a file, you can apply it against master with:

```
git apply patch-file.patch
```

**Comment by TomShawn** [comment]
> * correct sysvar doc information #11261

Thanks! I applied your diff to the doc and created https://github.com/pingcap/docs/pull/11273. PTAL
Some parts of the diff had already been updated in the upstream.

**Comment by TomShawn** [comment]
@morgo Is it time to close this issue now?

**Comment by morgo** [comment]
Sure, let's close for now. I'll open a new issue if I spot more.
